### ISSUE TITLE
Template behaves differently when redefining $data 

### ISSUE THREAD

**Issue by TurtleKwitty** [nuxt/nuxt#26239]
### Environment

- Operating System: Linux
- Node Version:     v21.6.2
- Nuxt Version:     3.10.3
- CLI Version:      3.10.1
- Nitro Version:    2.9.3
- Package Manager:  npm@10.2.4
- Builder:          -
- User Config:      devtools
- Runtime Modules:  -
- Build Modules:    -

### Reproduction

https://github.com/TurtleKwitty/NuxtPluginProvideBug

### Describe the bug

There is an unexpected interaction between plugin provided refs and local redefinitions with `const {$data} = useNuxtApp()`

When using the global object in a template a ref holding undefined or null will return {} and `.value` is required to make it behave correctly
When redefining it locally with `const {$data} = useNuxtApp()` then it is correctly returned as undefined or null

I will attempt to take a look if I can hunt down the cause of this bugged interaction but do not currently have a PR.

**Comment by warflash** [comment]
See -> https://github.com/nuxt/nuxt/issues/15709#issuecomment-1397417354

**Comment by TurtleKwitty** [comment]
> See -> [#15709 (comment)](https://github.com/nuxt/nuxt/issues/15709#issuecomment-1397417354)

Good point on it being a breaking change; it's the second time I've ran into it, but the first time a year ago I hadn't had time to investigate and couldn't find a reference as to why it was breaking. 

I'm thinking at least a reference to this unexpected interaction in the docs and possibly issuing a warning to the user would be a good middle ground to remove the confusion when one runs into this?

**Comment by pothuraju-dev** [comment]
Already a note is added in plugins docs
![image](https://github.com/nuxt/nuxt/assets/59384635/ff30a54f-d76c-43a7-a67c-16dd7bcdaf35)

I want to elaborate much about this issue in docs with the title **Providing reactive(ref/computed) as helpers** and showing an example with do ✅  and don't ❌ code snippets.

If this is okay I will do the changes in docs of plugins section.

**Comment by OrbisK** [comment]
@danielroe like @pothuraju-dev already mentioned, this is documented and default vue behavior:

https://github.com/nuxt/nuxt/blob/21b08c56fada0247b0051aec4fe3cd04922e4074/docs/2.directory-structure/1.app/1.plugins.md?plain=1#L206-L209

I think we should close it

**Comment by danielroe** [comment]
nuxtathon closed @OrbisK
